extensions/amp-selector/0.1/amp-selector.js

```diff
@@ -319,6 +341,54 @@ export class AmpSelector extends AMP.BaseElement {
     }
   }
 
+  /**
+   * Handles toggle action.
+   * @param {number} index
+   * @param {boolean} value
+   */
+  toggle_(index, value) {
+    // Change the selection to the next element in the specified direction.
+    // The selection should loop around if the user attempts to go one
+    // past the beginning or end.
+    const indexCurrentStatus = this.options_[index].hasAttribute('selected');
+    const indexFinalStatus =
+      !!(value !== undefined ? value : !indexCurrentStatus);
+    const selectedIndex = this.options_.indexOf(this.selectedOptions_[0]);
+
+    // There is a change of the `selected` attribute for the element
+    if (indexFinalStatus !== indexCurrentStatus) {
```

Comment:
Usually we use shortcut `if` statements to avoid nested `if`s, like

```js
if (indexFinalStatus === indexCurrentStatus) {
  return;
}

if (selectedIndex !== index) { /*...*/ } else { /*...*/ }